src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -738,12 +895,164 @@ export class SavedObjectsRepository {
       id,
       type,
       updated_at: time,
-      version: encodeHitVersion(response),
+      version: encodeHitVersion(updateResponse),
+      ...(this._registry.isMultiNamespace(type) && {
+        namespaces: updateResponse.get._source.namespaces,
+      }),
       references,
       attributes,
     };
   }
 
+  /**
+   * Adds one or more namespaces to a given multi-namespace saved object. This method and
+   * [`deleteFromNamespaces`]{@link SavedObjectsRepository.deleteFromNamespaces} are the only ways to change which Spaces a multi-namespace
+   * saved object is shared to.
+   */
+  async addToNamespaces(
+    type: string,
+    id: string,
+    namespaces: string[],
+    options: SavedObjectsAddToNamespacesOptions = {}
```

Comment:
I see only `addToNamespace` and `deleteFromNamespace`.

If an API wants to move a multi-NS SO from one NS to another, it would requires two calls? Don't we want a `setNamespaces` or `updateNamespaces(toAdd, toRemove)` to do it in a single call?

Reply:
Originally we had `updateNamespaces` but it was easier to reason about when breaking it down into two separate operations about. And it was much easier to deal with the authorization privilege checks (in `SecureSavedObjectsClientWrapper`) this way.

Reply:
I was thinking about this again, and it might be helpful to have some more context -- when I first implemented this, we would throw a Bad Request error if you tried to add a namespace that already existed in an object, or if you tried to remove a namespace that _didn't_ exist in an object.

However, I wound up changing that behavior during this review process so we simply allow those requests to succeed. From a pure `SavedObjectsRepository` point of view it might make sense to condense these into a single `updateNamespaces` operation. However, for the purposes of authorization, it's still much better for us to have two separate operations (otherwise we have to do a separate preflight operation within the `SecureSavedObjectsClientWrapper`).

Reply:
You have a better vision than me on the actual expected usages of these APIs. If you think current implementation is correct, I'm with you.